crates/mun_codegen_macros/src/lib.rs

```diff
@@ -213,8 +332,370 @@ pub fn as_value_derive(input: TokenStream) -> TokenStream {
         Data::Union(_) => {
             unimplemented!("#[derive(AsValue)] is not defined for unions!");
         }
-        Data::Enum(_) => {
-            unimplemented!("#[derive(AsValue)] is not defined for enums!");
+        Data::Enum(enum_data) => {
+            const SUPPORTED_TAG_SIZES: &[&str] =
```

Comment:
Does it make sense to split these two into sperate functions? One for struct and one for enum?

Reply:
`proc_macro` is a wrapper around `proc_macro2`. `proc_macro2` is unstable. `proc_macro` is stable. The results of `quote` etc. are all `proc_macro2`, hence why you need to call `.into()`.

That's all legal because you are not actually importing unstable types. However, once you use it as function parameters though, you have to import those types - which is not possible iirc.